recipes/tests.rb

```diff
@@ -138,3 +138,68 @@
     JQMERGE
   end
 end
+
+# only test on GPU instances
+if node['cfncluster']['nvidia']['enabled'] == 'yes'
+  bash 'test nvidia driver install' do
+    cwd Chef::Config[:file_cache_path]
+    code <<-TESTNVIDIA
+      has_gpu=$(lspci | grep -o "NVIDIA")
+      if [ -z $has_gpu ]; then
+        echo "No GPU detected, no test needed."
+        exit 0
+      fi
+
+      set -e
+      driver_ver="418.87"     #Set this to NVIDIA driver version that we are currently installing
+      export PATH="/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/opt/aws/bin"
+
+      # Test NVIDIA Driver installation
+      echo "Testing NVIDIA driver install..."
+      # grep driver version from nvidia-smi output. If driver is not installed nvidia-smi command will fail
+      driver_output=$(nvidia-smi | grep -E -o "Driver Version: [0-9]*.[0-9]*")
+      if [ "$driver_output" != "Driver Version: $driver_ver" ]; then
+        echo "NVIDIA driver installed incorrectly! Installed $driver_output but expected version $driver_ver"
+        exit 1
+      else
+        echo "Correctly installed NVIDIA $driver_output"
+      fi
+    TESTNVIDIA
+  end
+end
+
+# only test on GPU instances
+if node['cfncluster']['nvidia']['enabled'] == 'yes'
```

Comment:
Can these blocks be combined?

Reply:
fdm wanted them separate, so we can see which block(driver or CUDA) is failing in kitchen test

Reply:
you can put the two "bash" resources in the same "if node['cfncluster']['nvidia']['enabled'] == 'yes'"